### DISCUSSION TITLE
Why is the server CPM inconsistent with the client CPM on the topology page

### DISCUSSION THREAD

**Discussion by BertOnline** [apache/skywalking#8449] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/17882970/150105825-69b6e18e-368c-475f-863b-490e3cb79fa3.png)
![image](https://user-images.githubusercontent.com/17882970/150105871-2a4f7d38-0a45-4238-919a-42965ea8b61d.png)
Why is the server CPM inconsistent with the client CPM on the topology page? This is a sample for the same time period.

**Comment by wu-sheng** [comment, marked as answer]
This is not a question that the community should answer.
But I am just giving you some clues. Take a look at your percentile metric, which is very unstable, which means, your client(s) are accessing the server(s) in high concurrency, but the server HTTP/RPC handlers are blocked from time to time. But due to your timeout setting is not reached, all requests are eventually processed but in different time buckets.

This is why we provided 2 metrics called client-side and server-side. In most applications, it isn't as consistent as people thought. This is the real world(Excluding the plugin bug).

____

On the other side, if a server-side plugin has a bug or conflict(just if), then a request could be counted more than once, such as `undertow + SpringMVC` may have this case, and count a request twice due to across threads accessing.

___

Either way, the community can't and shouldn't debug for you. Once you are using an open-source project, you need to take more time on your own and try to collaborate.
